frame/child-bounties/src/lib.rs

```diff
@@ -0,0 +1,892 @@
+	#[pallet::storage]
+	#[pallet::getter(fn parent_child_bounties)]
+	pub type ParentChildBounties<T: Config> =
+		StorageMap<_, Twox64Concat, BountyIndex, BountyIndex, ValueQuery>;
+
+	/// Child-bounties that have been added.
+	#[pallet::storage]
+	#[pallet::getter(fn child_bounties)]
+	pub type ChildBounties<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		BountyIndex,
+		Twox64Concat,
+		BountyIndex,
+		ChildBounty<T::AccountId, BalanceOf<T>, T::BlockNumber>,
+	>;
+
+	/// The description of each child-bounty.
+	#[pallet::storage]
+	#[pallet::getter(fn child_bounty_descriptions)]
+	pub type ChildBountyDescriptions<T: Config> =
+		StorageMap<_, Twox64Concat, BountyIndex, BoundedVec<u8, T::MaximumReasonLength>>;
+
+	/// The cumulative child-bounty curator fee for each parent bounty.
+	#[pallet::storage]
+	#[pallet::getter(fn children_curator_fees)]
+	pub type ChildrenCuratorFees<T: Config> =
+		StorageMap<_, Twox64Concat, BountyIndex, BalanceOf<T>, ValueQuery>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Add a new child-bounty.
+		///
+		/// The dispatch origin for this call must be master curator.
+		/// parent bounty must be in "active" state.
+		///
+		/// Child-bounty gets added successfully & fund gets transferred from
+		/// parent bounty to child-bounty account, if parent bounty has
+		/// enough fund. Else call gets failed.
+		///
+		/// Upper bound to maximum number of active  child-bounties that
+		/// can be added are managed via runtime trait config
+		/// 'MaxActiveChildBountyCount'.
+		///
+		/// If the call is success, the state of child-bounty is
+		/// moved to "Added" state.
+		///
+		/// - `bounty_id`: Bounty ID for which child-bounty to be added.
+		/// - `value`: Value for executing the proposal.
+		/// - `description`: Text description for the child-bounty.
+		#[pallet::weight(<T as Config>::WeightInfo::add_child_bounty(description.len() as u32))]
+		pub fn add_child_bounty(
+			origin: OriginFor<T>,
+			#[pallet::compact] parent_bounty_id: BountyIndex,
+			#[pallet::compact] value: BalanceOf<T>,
+			description: Vec<u8>,
+		) -> DispatchResult {
+			let signer = ensure_signed(origin)?;
+
+			// Verify the arguments.
+			let bounded_description =
+				description.try_into().map_err(|_| BountiesError::<T>::ReasonTooBig)?;
+			ensure!(value >= T::ChildBountyValueMinimum::get(), BountiesError::<T>::InvalidValue);
+			ensure!(
+				Self::parent_child_bounties(parent_bounty_id) <=
+					T::MaxActiveChildBountyCount::get() as u32,
+				Error::<T>::TooManyChildBounties,
+			);
+
+			let (curator, _) = Self::ensure_bounty_active(parent_bounty_id)?;
+			ensure!(signer == curator, BountiesError::<T>::RequireCurator);
+
+			// Read parent bounty account info.
+			let parent_bounty_account =
+				pallet_bounties::Pallet::<T>::bounty_account_id(parent_bounty_id);
+
+			// Ensure parent bounty has enough balance after adding child-bounty.
+			let balance = T::Currency::free_balance(&parent_bounty_account);
+			ensure!(
```

Comment:
can do this with `ensure_can_withdraw` and ensuring that the source account lives afterwards.

Reply:
Ping.

Reply:
Done. Using `ensure_can_withdraw` and afterwards the actual transfer now uses `KeepAlive` to make sure the account lives.